### PR TITLE
Download exhibition catalogue by admin

### PR DESCRIPTION
**Pull request by Mouhamed114**

### CODE REVIEW

**Review by MisterPingouin — APPROVED** [review]

File: public/downloadPdf/Download.pdf
Comment (aldaitzwild) [inline public/downloadPdf/Download.pdf:1]: This file shouldn't be versionned

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

File: src/Service/CroppedService.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        $this->twig = $twig;
+        $this->pictureRepository = $pictureRepository;
+        $this->parameterBag = $parameterBag;
+    }
+    public function cropped(Picture $picture): Response
+    {
+        $html =  $this->twig->render('picture/cropped.html.twig', [
+            'picture' => $picture,
+        ]);
+        return new Response($html);
```
Comment (aldaitzwild) [inline src/Service/CroppedService.php]: The "return new Response(...)" is the responsibility of the controller, it should not be in a service.

File: src/Service/CroppedService.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            throw new NotFoundHttpException('Image not found');
+        }
+
+        $picture->setImageCrop('uploads/images/' . $filename);
+        $this->pictureRepository->save($picture, true);
+        return new Response(
+            json_encode(['success' => true]),
+            Response::HTTP_OK,
+            ['Content-Type' => 'application/json']
+        );
```
Comment (aldaitzwild) [inline src/Service/CroppedService.php]: The "return new Response(...)" is the responsibility of the controller, it should not be in a service.

File: src/Service/GeneratePdfService.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        $pictures = $this->pictureRepository->findAll();
+        $html = $this->twig->render('pdf/generatePdfExhibition.html.twig', [
+            'pictures' => $pictures,
+        ]);
+        $pdf = $this->knpSnappy->getOutputFromHtml($html);
+        $pdfFilePath = 'downloadPdf/Download.pdf';
+        file_put_contents($pdfFilePath, $pdf);
+        $response = new BinaryFileResponse($pdfFilePath);
+        $response->setContentDisposition(ResponseHeaderBag::DISPOSITION_ATTACHMENT, 'Download.pdf');
+        return $response;
```
Comment (aldaitzwild) [inline src/Service/GeneratePdfService.php]: The "return new Response(...)" is the responsibility of the controller, it should not be in a service.

File: src/Service/GeneratePdfService.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    public function downloadPdf(): BinaryFileResponse
+    {
+        $pdfFilePath = 'downloadPdf/Download.pdf';
+        $response = new BinaryFileResponse($pdfFilePath);
+        $response->setContentDisposition(ResponseHeaderBag::DISPOSITION_ATTACHMENT, 'Download.pdf');
+        return $response;
```
Comment (aldaitzwild) [inline src/Service/GeneratePdfService.php]: The "return new Response(...)" is the responsibility of the controller, it should not be in a service.

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

**Review by aldaitzwild — APPROVED** [review]

**Pull request merged**